admin-dev/themes/new-theme/js/pages/order/view/order-product-manager.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import Router from '../../../components/router';
+import {EventEmitter} from '../../../components/event-emitter';
+import OrderViewEventMap from '../view/order-view-event-map';
+
+const $ = window.$;
+
+export default class OrderProductManager {
+  constructor() {
+    this.router = new Router();
+  }
+
+  handleDeleteProductEvent(event) {
+    event.preventDefault();
+
+    const $btn = $(event.currentTarget);
+
+    const confirmed = confirm($btn.data('delete-message'));
+
+    if (!confirmed) {
+      return;
+    }
+
+    this.deleteProduct($btn.data('order-id'), $btn.data('order-detail-id'));
+  }
+
+  deleteProduct(orderId, orderDetailId) {
+    $.ajax(this.router.generate('admin_orders_delete_product', {orderId, orderDetailId}), {
+      method: 'POST',
+    }).then(() => {
+      EventEmitter.emit(OrderViewEventMap.productDeletedFromOrder, {
+        oldOrderDetailId: orderDetailId,
+        orderId,
+      });
+    });
+  }
+
+  paginate(orderId, numPage) {
```

Comment:
Same here: in the future we might we want to extract this logic into a JS Paginator component we could apply everywhere. The constructor would take the necessary arguments such as route, number of rows ...

Reply:
The right way should have been to use the component Grid (and evolve it), and not recreate a new paginator.

Reply:
But @matks explains me we can't use grid component in this case because data were loaded (and reloaded) in AJAX.

Reply:
@jolelievre Could you create an issue for this ? 
May be we could refactor this in 1.7.8 with help of our new front end developper @NeOMakinG and use a grid for products ? #continuousImprovement

Reply:
I'll create the issue

Reply:
https://github.com/PrestaShop/PrestaShop/issues/16688